framework/src/modules/interoperability/mainchain/store.ts

```diff
@@ -12,9 +12,96 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
-//  TODO: extends BaseInteroperabilityStore
-// eslint-disable-next-line  @typescript-eslint/no-extraneous-class
-export class MainchainInteroperabilityStore {
-	// public async forward(ccu: CCUpdateParams, ccm: CCMsg): Promise<void> {};
-	// public async bounce(ccu: CCUpdateParams, ccm: CCMsg): Promise<void> {};
+import { BaseInteroperabilityStore } from '../base_interoperability_store';
+import { LIVENESS_LIMIT } from '../constants';
+import { CCMsg, CCUpdateParams, SendInternalContext } from '../types';
+
+export class MainchainInteroperabilityStore extends BaseInteroperabilityStore {
+	public async isLive(chainID: Buffer, timestamp: number): Promise<boolean> {
+		try {
+			const terminatedStateAccount = await this.getTerminatedStateAccount(chainID);
```

Comment:
[idea] if we need to check existence, it would be better to add a function specific to that. `get` and `has` will have different performance, and it's more explicit.